packages/desktop-client/src/components/accounts/TransactionsTable.js

```diff
@@ -1787,6 +1837,120 @@ export let TransactionTable = forwardRef((props, ref) => {
     id => splitsExpanded.dispatch({ type: 'toggle-split', id }),
     [splitsExpanded.dispatch],
   );
+  let [field, setField] = useState(null);
+  let [ascDesc, setAscDesc] = useState('desc');
+
+  function onSortTable(e) {
+    let transform =
+      e.currentTarget.innerText.toLowerCase() === 'payment' ||
+      e.currentTarget.innerText.toLowerCase() === 'deposit'
+        ? 'amount'
+        : e.currentTarget.innerText.toLowerCase();
+    if (transform === field) {
+      ascDesc === 'desc' ? setAscDesc('asc') : setAscDesc('desc');
+    } else {
+      setField(transform);
+      transform === 'amount' || transform === 'date'
+        ? setAscDesc('asc')
+        : setAscDesc('desc');
+    }
+  }
+
+  let category = props.categoryGroups.flatMap(cat => {
+    return cat.categories;
+  });
+  let fields = [field] + 's';
+
+  function sortStrings() {
+    transactions =
+      ascDesc === 'asc'
```

Comment:
Instead of having two different behaviors here for ascending and descending, couldn’t you always pick one way to sort, and then call `reverse` on the array when you’re done if needed?

Reply:
Taking me to school 🚌, I love it! This worked great. thanks for the suggestion.